content/articles/how-to-ship-zero-javascript-with-astro/index.md

```diff
@@ -0,0 +1,464 @@
+ 
+Before we go further. Type in the command below in the terminal. This command installs the required packages(dependencies):
+ 
+```
+ 
+npm install
+ 
+```
+ 
+### Step 2 - Application Structure
+ 
+In this step, we will have an overview of what our application folder structure looks like.
+ 
+We have different folders such as:
+ 
+* src
+ 
+* components
+ 
+* layouts
+ 
+* pages
+ 
+* public
+ 
+* package.json
+ 
+The ones we are more interested in are:
+ 
+* **src**: It contains the components, layout, and pages folder. The src is where our project source codes will be.
+ 
+* **components**: This is inside the src folder. The components folder will contain our React UI components.
+ 
+* **pages**: This is inside the src folder. Pages folder includes the **index.astro**, this serves as our entry point
+ 
+* **astro.config.mjs**: This is a configuration file for our Astro project. It contains a configuration object.
+ 
+```
+ 
+├── src/
+ 
+│   ├── components/
+ 
+|   |       |── Tours.astro
+ 
+│   ├── layouts/
+ 
+│   └── pages/
+ 
+│       └── index.astro
+ 
+├── public/
+ 
+├── astro.config.mjs
+ 
+└── package.json
+ 
+```
+ 
+### Step 3 - Installing and Configuring Our Framework to Work With Astro
+ 
+The framework we will be using is React. To make Astro support React run the command below. It installs React renderer and React.
+ 
+```
+ 
+npm install @astrojs/renderer-react react react-dom -D
+ 
+```
+ 
+After the installation of React renderer and React. Navigate to **astro.config.mjs** add the code below to the configuration object.
+ 
+```
+ 
+export default {
+ 
+renderers: ['@astrojs/renderer-react'],
+ 
+```
+ 
+A renderer does the following:
+ 
+* Transforms your component into strings of HTML.
+* Rehydrates the HTML once it reaches the browser.
+ 
+### step 4 - Starting Our Development Server
+ 
+Start the development server with the command below:
+ 
+```
+ 
+npm start
+ 
+```
+ 
+When you start the project for the first time. Astro, along with Snowpack will prepare your dependencies. It will then start the development server on http://localhost:300/.
+ 
+You can navigate to http://localhost:300/ to access your Astro project. This will show some helpful information and a welcome message like the image below:
+ 
+![new-astro-project](/engineering-education/how-to-ship-zero-javascript-with-astro/new-astro-app-scaled.jpg)
+ 
+### Step 5 -Creating a New React Component in Astro
+ 
+In this step, you will create a React Component. A React component is one of the main building blocks of React. It is written in **jsx** format. It tells React what should be rendered on the screen.
+ 
+There are two types of React components:
+ 
+* class component
+ 
+* functional component
+ 
+For this article, we will be using a functional component.
+ 
+Navigate to the component folder in the main project folder. Create a **Tour.jsx** file in the component folder.
+ 
+Inside the **Tour.jsx** file you created add the following:
+ 
```

Comment:
You made the changes but you didn't fix this typo. You've stated that its *Tour.jsx* while the rest of the steps you referring to it as *Tours.jsx*.